src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -2316,3 +2316,10 @@ OperatorImplementsEquality(Oid opno)
 
 	return equalityOperator;
 }
+
+
+bool
+QueryContainsDistributedTableRTE(Query *query)
+{
+	return ListContainsDistributedTableRTE(ExtractRangeTableEntryList(query), NULL);
```

Comment:
maybe `FindNodeMatchingCheckFunction((Node *) subquery, IsDistributedTableRTE))`

Reply:
Thanks, was looking for such convenient routines.